src/debugAdapter/goDebug.ts

```diff
@@ -995,49 +995,67 @@ class GoDebugSession extends DebugSession {
 	protected variablesRequest(response: DebugProtocol.VariablesResponse, args: DebugProtocol.VariablesArguments): void {
 		verbose('VariablesRequest');
 		let vari = this._variableHandles.get(args.variablesReference);
-		let variables;
+		let variablesPromise: Promise<DebugProtocol.Variable[]>;
+		let loadChildren = async (exp: string, v: DebugVariable) => {
+			if (v.len > 0 && v.children.length === 0) {
+				// from https://github.com/derekparker/delve/blob/master/Documentation/api/ClientHowto.md#loading-more-of-a-variable
+				await this.evaluateRequestImpl({ 'expression': exp }).then(result => {
+					const variable = this.delve.isApiV1 ? <DebugVariable>result : (<EvalOut>result).Variable;
+					v.children = variable.children;
+				});
+			}
+		};
 		if (vari.kind === GoReflectKind.Array || vari.kind === GoReflectKind.Slice) {
-			variables = vari.children.map((v, i) => {
-				let { result, variablesReference } = this.convertDebugVariableToProtocolVariable(v, i);
-				return {
-					name: '[' + i + ']',
-					value: result,
-					evaluateName: vari.fullyQualifiedName + '[' + i + ']',
-					variablesReference
-				};
-			});
+			variablesPromise = Promise.all(vari.children.map(async (v, i) => {
+					await loadChildren(`*(*${v.type})(${v.addr})`, v);
+					let { result, variablesReference } = this.convertDebugVariableToProtocolVariable(v);
+					return {
+						name: '[' + i + ']',
+						value: result,
+						evaluateName: vari.fullyQualifiedName + '[' + i + ']',
+						variablesReference
+					};
+				})
+			);
 		} else if (vari.kind === GoReflectKind.Map) {
-			variables = [];
-			for (let i = 0; i < vari.children.length; i += 2) {
-				if (i + 1 >= vari.children.length) {
-					break;
+			variablesPromise = Promise.resolve((async () => {
+				let variables = new Array<DebugProtocol.Variable>();
+				for (let i = 0; i < vari.children.length; i += 2) {
+					if (i + 1 >= vari.children.length) {
+						break;
+					}
+					let mapKey = this.convertDebugVariableToProtocolVariable(vari.children[i]);
+					await loadChildren(`${vari.fullyQualifiedName}.${vari.name}[${mapKey.result}]`, vari.children[i + 1]);
+					let mapValue = this.convertDebugVariableToProtocolVariable(vari.children[i + 1]);
+					variables.push({
+						name: mapKey.result,
+						value: mapValue.result,
+						evaluateName: vari.fullyQualifiedName + '[' + mapKey.result + ']',
+						variablesReference: mapValue.variablesReference
+					});
 				}
-				let mapKey = this.convertDebugVariableToProtocolVariable(vari.children[i], i);
-				let mapValue = this.convertDebugVariableToProtocolVariable(vari.children[i + 1], i + 1);
-				variables.push({
-					name: mapKey.result,
-					value: mapValue.result,
-					evaluateName: vari.fullyQualifiedName + '[' + mapKey.result + ']',
-					variablesReference: mapValue.variablesReference
-				});
-			}
+				return variables;
+			})());
```

Comment:
That was my original implementation, I switched it though so that it's uniform with the other blocks.  I don't feel super strong about this though so if you think the other way is more clear I can change it.

Reply:
I mean i think its easier to read if we put on name on it, otherwise ill spend a while reading it to understand what exactly it does